bulb.py

```diff
@@ -0,0 +1,59 @@
+
+'''Class that contains the set of rules around sensors'''
+class Bulb(object):
+
+    def __init__(self,  name, lightSensor, presenceSensors, homeSensor, wastage_function):
+        self.name = name
+        self.light_sensor = lightSensor
+        self.presence_sensors = presenceSensors
+        self.home_sensor = homeSensor
+        self.wastage_function = wastage_function
+
+
+    def isWasted(self, *args):
+        return self.wastage_function(self,*args)
+
+
+# Actual function implementation for the given lights
+
+def kitchenLampWasted(self):
```

Comment:
Yes, it is desired. In order to pass the function as a parameter it can't be a part of the actual bulb object (could just make it a static method now that I think about it..). The 'self' argument is in this case just an arbitraty name, but when used on the object it does in fact refer to the object.